### PR TITLE
Add radio list

### PR DESCRIPTION
**Pull request by bmigirl**
This will only be in checkout temporarily, until we have connected it with UI kit that is currently in development. 
- Add radio and radio group
- Add some utils for address fields that'll be used in the next pr when displaying addresses with the radio boxes
- Add some queries and mutations (not sure how they got to this pr, most likely some code leftover but they'll be needed in the next one so letting them be)

### CODE REVIEW

File: apps/checkout/src/lib/utils/address.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    if (!addressFields.includes(orderedAddressField)) {
+      return result;
+    }
+
+    return [...result, orderedAddressField];
+  }, [] as AddressField[]);
+
+export const getSortedAddressFieldsFromAddress = (
+  address: Partial<Record<AddressField, any>>
+) => getSortedAddressFields(Object.keys(address) as AddressField[]);
```
Comment (jwm0) [inline apps/checkout/src/lib/utils/address.ts:24]: The implementation would be a bit cleaner if we didn't map it to an array as the middle step, if `getSortedAddressFields` isn't useful on it's own I'd skip it and make it a single function:
```
export const getSortedAddressFields = (address: Partial<Record<AddressField, any>>) =>
  addressFieldsOrder.reduce((result, orderedAddressField) => {
    if (orderedAddressField in address) {
      return result;
    }

    return [...result, orderedAddressField];
  }, [] as AddressField[]);
```

File: apps/checkout/src/sections/CheckoutForm.tsx [hunk 2]
```diff
@@ -24,14 +29,41 @@ export const CheckoutForm = () => {
...
+          <Divider className="my-4" />
+          <RadioGroup
+            label="lolz"
+            alignment="top"
+            onSelect={setSelectedValue}
+            selectedValue={selectedValue}
+            options={[
+              {
+                id: "address",
+                // @ts-ignore for development
```
Comment (jwm0) [inline apps/checkout/src/sections/CheckoutForm.tsx:60]: .

File: apps/checkout/src/sections/CheckoutForm.tsx [hunk 3]
```diff
@@ -24,14 +29,41 @@ export const CheckoutForm = () => {
   const methods = useForm<FormData>({ resolver, mode: "onBlur" });
   const { setValue, watch } = methods;
 
+  const [selectedValue, setSelectedValue] = useState<string>();
+
   const handleEmailChange = (value: string) => setValue("email", value);
 
+  const addressData: Partial<AddressFragment> = {
```
Comment (jwm0) [inline apps/checkout/src/sections/CheckoutForm.tsx:36]: you might want to use some fake data for that

File: apps/checkout/src/components/RadioGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { Radio, RadioOption, RadioProps } from "./Radio";
+
+interface RadioGroupProps
+  extends Pick<RadioProps, "onSelect" | "selectedValue" | "alignment"> {
+  options: RadioOption[];
+  label: string;
+}
+
+export const RadioGroup: React.FC<RadioGroupProps> = ({
```
Comment (jwm0) [inline apps/checkout/src/components/RadioGroup.tsx:10]: I know this is temporary but let's use compound components pattern for this one in the ui kit, cc @adrian-potepa

File: apps/checkout/src/lib/utils/address.ts (hunk 1 above)
Comment (bmigirl) [reply apps/checkout/src/lib/utils/address.ts:24]: I think `getSortedAddressFields` will be used on it's own once we add it to the address form, where we get address fields as string[] straight from the api

File: apps/checkout/src/sections/AddressBoxContent.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  onEdit,
+}) => {
+  const name = address.name || `${address.firstName} ${address.lastName}`;
+
+  return (
+    <div className="flex flex-row justify-between w-full">
+      <div className="flex flex-col">
+        <label htmlFor={htmlFor}>
+          <Text weight="semibold">{name}</Text>
+          {without(getSortedAddressFieldsFromAddress(address), "name").map(
```
[the remark this answers is not in the document]
Comment (bmigirl) [reply apps/checkout/src/sections/AddressBoxContent.tsx:31]: Oh yeah, that's to be removed actually, good catch

File: apps/checkout/src/sections/CheckoutForm.tsx (hunk 3 above)
Comment (bmigirl) [reply apps/checkout/src/sections/CheckoutForm.tsx:36]: It's just for preview purposes for this pr, we can just remove it altogether in the next pr

File: apps/checkout/src/sections/AddressBoxContent.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (bmigirl) [reply apps/checkout/src/sections/AddressBoxContent.tsx:31]: Actually this was here temporarily, removing

File: apps/checkout/src/sections/AddressBoxContent.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import { getSortedAddressFieldsFromAddress } from "@lib/utils/address";
+import { Text } from "@components/Text";
+import React from "react";
+import { without } from "lodash";
```
[the remark this answers is not in the document]
Comment (bmigirl) [reply apps/checkout/src/sections/AddressBoxContent.tsx:4]: As above, removing

File: apps/checkout/src/components/RadioGroup.tsx (hunk 4 above)
Comment (bmigirl) [reply apps/checkout/src/components/RadioGroup.tsx:10]: Okay

File: apps/checkout/src/sections/Contact/Contact.tsx [hunk 7]
```diff
@@ -46,11 +46,12 @@ export const Contact = ({ onEmailChange, email }: ContactProps) => {
       setCurrentSection("signedInUser");
 
       if (checkout?.user?.id !== user?.id) {
-        customerAttach(
-          getDataWithToken({
-            customerId: user?.id as string,
-          })
-        );
+        // TMP for development
```
[the remark this answers is not in the document]
Comment (bmigirl) [reply apps/checkout/src/sections/Contact/Contact.tsx:49]: .

File: apps/checkout/src/sections/AddressBoxContent.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  const name = address.name || `${address.firstName} ${address.lastName}`;
+
+  return (
+    <div className="flex flex-row justify-between w-full">
+      <div className="flex flex-col">
+        <label htmlFor={htmlFor}>
+          <Text weight="semibold">{name}</Text>
+          {getSortedAddressFieldsFromAddress(address).map(
+            (field: AddressField) => (
+              <Text>{address[field] as string}</Text>
```
Comment (orzechdev) [inline apps/checkout/src/sections/AddressBoxContent.tsx:32]: Missing key in Text component

Comment (jwm0) [reply apps/checkout/src/sections/AddressBoxContent.tsx]: We should add a linter rule to catch that

**Review by orzechdev — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**